src/robotFactory.js

```diff
@@ -1,10 +1,71 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y = this.coords.y + step;
+  }
+  goBack(step = 1) {
+    this.coords.y = this.coords.y - step;
+  }
+  goRight(step = 1) {
+    this.coords.x = this.coords.x + step;
+  }
+  goLeft(step = 1) {
+    this.coords.x = this.coords.x - step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${
+      this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords.z = coords.z || 0;
+  }
+
+  goUp(step = 1) {
+    this.coords.z = this.coords.z + step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z = this.coords.z - step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords,
+    chipVersion, maxLoadWeight, currentLoad = null) {
```

Comment:
if you move one argument to the new line - you move each of them to the new line
```suggestion
  constructor(
    name, 
    weight, 
    coords,
    chipVersion,
    maxLoadWeight, 
    currentLoad = null
  ) {
```